contracts/test/PDPService.t.sol

```diff
@@ -289,17 +311,21 @@ contract PDPServiceProofSetMutateTest is Test {
         roots[0] = PDPService.RootData(Cids.Cid(abi.encodePacked("test")), 64);
         pdpService.addRoots(setId, roots);
         recordAssert.expectRecord(PDPListener.OperationType.ADD, setId);
+        assertEq(pdpService.getNextChallengeEpoch(setId), block.number + challengeFinalityDelay);
 
         // Remove root
         uint256[] memory toRemove = new uint256[](1);
         toRemove[0] = 0;
-        pdpService.removeRoots(setId, toRemove);
-        recordAssert.expectRecord(PDPListener.OperationType.REMOVE, setId);
+        pdpService.scheduleRemovals(setId, toRemove);
+        recordAssert.expectRecord(PDPListener.OperationType.REMOVE_SCHEDULED, setId);
```

Comment:
I think we need to explicitly check the extra data associated with the event. It's a critical part of correctness for applications built on this.

Reply:
Consider asserting that the root is not yet removed in any of the methods where this could be observed.

Reply:
I think you're saying this is important for all of our expectRecord checks.  If so I agree.  But I'll wait for #55 and build a framework `expectScheduleRemove` `expectProvePossesion` etc.